taichi/ir/frontend_ir.cpp

```diff
@@ -192,7 +192,8 @@ void BinaryOpExpression::type_check(CompileConfig *config) {
     ret_type = PrimitiveType::i32;
     return;
   }
-  if (is_shift_op(type)) {
+  if (is_shift_op(type) ||
```

Comment:
aha we may gradually add a table in docs/spec to show the expected type of different ops.

Reply:
Yep. It is currently described here: https://docs.taichi-lang.org/docs/type#implicit-type-casting-in-binary-operations
For `pow()` I think I may need to add something in the API reference, which will be done in a follow-up PR.